packages/blocks/table/src/Table.tsx

```diff
@@ -34,51 +44,105 @@ type AppProps = {
   entityId: string;
 };
 
-const defaultData: AppProps["data"] = { data: [] };
+const useTableData = (
+  linkedAggregation: BlockProtocolLinkedAggregation | undefined,
+) => {
+  const defaultTableData: TableData = {
+    linkedAggregation,
+    data: linkedAggregation?.results,
+  };
 
-const useTableData = (data: AppProps["data"]) => {
   const defaultAggregateData = {
-    tableData: data,
-    prevTableData: data,
+    tableData: defaultTableData,
+    prevLinkedAggregation: linkedAggregation,
   };
 
-  const [{ prevTableData, tableData }, setAggregateTableData] =
+  const [{ prevLinkedAggregation, tableData }, setAggregateTableData] =
     useState(defaultAggregateData);
 
   const setTableData = useCallback(
-    (nextData: AppProps["data"]) =>
+    (nextData: TableData) =>
       setAggregateTableData((existing) => ({
         ...existing,
         tableData: nextData,
       })),
     [],
   );
 
-  if (data !== prevTableData) {
+  if (linkedAggregation !== prevLinkedAggregation) {
     setAggregateTableData(defaultAggregateData);
   }
 
   return [tableData, setTableData] as const;
 };
 
-export const App: BlockComponent<AppProps> = ({
-  data = defaultData,
+const getLinkedAggregation = (params: {
+  linkedAggregations: BlockProtocolLinkedAggregation[];
+  path: string;
+  sourceEntityId: string;
+}): BlockProtocolLinkedAggregation | undefined => {
+  const { linkedAggregations, path, sourceEntityId } = params;
+
+  return linkedAggregations.find(
+    (aggregation) =>
+      aggregation.path === path &&
+      aggregation.sourceEntityId === sourceEntityId,
+  );
+};
+
+const cleanUpdateLinkedAggregationAction = (
+  action: BlockProtocolUpdateLinksAction & {
+    updatedOperation: Partial<BlockProtocolLinkedAggregation> & {
+      __typename?: string;
+    };
+  },
+) => {
+  return produce(action, (draftAction) => {
+    draftAction.updatedOperation.multiSort =
+      action.updatedOperation.multiSort?.map((sort) => {
```

Comment:
You're still mutating the passed action here – you should map over the drafted action to prevent this

```suggestion
      draftAction.updatedOperation.multiSort?.map((sort) => {
```

Reply:
Thanks Nate! Added this in.